packages/material-ui/src/Slider/Slider.js

```diff
@@ -679,6 +697,16 @@ const Slider = React.forwardRef(function Slider(props, ref) {
     };
   }, [handleTouchEnd, handleTouchMove, handleTouchStart]);
 
+  React.useEffect(() => {
+    if (disabled) {
+      const doc = ownerDocument(sliderRef.current);
+      doc.removeEventListener('mousemove', handleTouchMove);
+      doc.removeEventListener('mouseup', handleTouchEnd);
+      doc.removeEventListener('touchmove', handleTouchMove);
+      doc.removeEventListener('touchend', handleTouchEnd);
+    }
+  }, [disabled, handleTouchEnd, handleTouchMove]);
+
```

Comment:
Not sure about the duplication of the event listener logic, it feels brittle to future changes.

What if we restored the previous approach but with a new branch like:

```jsx
if (disabled) {
  return undefined;
}
```

Reply:
Don't try to be clever with effects. The useEffect hook has cleanup logic built in for a reason. You're abusing cleanup methods for something that isn't added in the effect.

Reply:
I was thinking of going for:

```diff
diff --git a/packages/material-ui/src/Slider/Slider.js b/packages/material-ui/src/Slider/Slider.js
index 35a62d356b..73d0fa44b1 100644
--- a/packages/material-ui/src/Slider/Slider.js
+++ b/packages/material-ui/src/Slider/Slider.js
@@ -660,6 +660,10 @@ const Slider = React.forwardRef(function Slider(props, ref) {
   });

   React.useEffect(() => {
+    if (disabled) {
+      return undefined;
+    }
+
     const { current: slider } = sliderRef;
     slider.addEventListener('touchstart', handleTouchStart, {
       passive: doesSupportTouchActionNone(),
@@ -677,7 +681,7 @@ const Slider = React.forwardRef(function Slider(props, ref) {
       doc.removeEventListener('touchmove', handleTouchMove);
       doc.removeEventListener('touchend', handleTouchEnd);
     };
-  }, [handleTouchEnd, handleTouchMove, handleTouchStart]);
+  }, [disabled, handleTouchEnd, handleTouchMove, handleTouchStart]);

   const handleMouseDown = useEventCallback((event) => {
     if (onMouseDown) {
```

What do you feel as cleaver about it? disabled => no listeners